pkg/server/tls_windows.go

```diff
@@ -0,0 +1,342 @@
+package server
+
+import (
+	"crypto"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/asn1"
+	"fmt"
+	"regexp"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/github/smimesign/certstore"
+)
+
+func (l *tlsListener) applyWindowsCertificateStore(c TLSConfig) error {
+
+	// Restrict normal TLS options when using windows certificate store
+	if c.TLSCertPath != "" {
+		return fmt.Errorf("cannot include certificate file when using windows certificate store")
+	}
+	if c.TLSKeyPath != "" {
+		return fmt.Errorf("cannot include key file when using windows certificate store")
+	}
+
+	var subjectRegEx *regexp.Regexp
+	var err error
+	if c.WindowsCertificateFilter.ClientSubjectRegEx != "" {
+		subjectRegEx, err = regexp.Compile(c.WindowsCertificateFilter.ClientSubjectRegEx)
+		if err != nil {
+			return fmt.Errorf("error compiling subject common name regular expression %w", err)
+		}
+	}
+
+	// If there is an existing windows certhandler notify it to stop refreshing
+	if l.windowsCertHandler != nil {
+		l.windowsCertHandler.stopUpdateTimer <- struct{}{}
+	}
+
+	cn := &winCertStoreHandler{
+		cfg:             *c.WindowsCertificateFilter,
+		subjectRegEx:    subjectRegEx,
+		stopUpdateTimer: make(chan struct{}, 1),
+	}
+	err = cn.refreshCerts()
+	if err != nil {
+		return err
+	}
+	certPool := x509.NewCertPool()
+	certPool.AddCert(cn.clientRootCA)
+	config := &tls.Config{
+		VerifyPeerCertificate: cn.VerifyPeer,
+		GetCertificate: func(info *tls.ClientHelloInfo) (*tls.Certificate, error) {
+			cn.winMut.Lock()
+			defer cn.winMut.Unlock()
+			cert := &tls.Certificate{
```

Comment:
Do we want to set `Leaf` here?

```go
// Leaf is the parsed form of the leaf certificate, which may be initialized
// using x509.ParseCertificate to reduce per-handshake processing. If nil,
// the leaf certificate will be parsed as needed.
Leaf *x509.Certificate
```

It might make handshakes a bit faster.